streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -725,6 +725,33 @@ trait ZStream[-R, +E, +A] extends Serializable { self =>
         }
     }
 
+  def split(p : A => UIO[Boolean]) : ZManaged[R, E, (ZStream[R,E,A],ZStream[R,E,A])] = 
+    split(2)(p)
+
+  def split[B1,B2](queueCapacity : Int)(p : A => UIO[Boolean]) : ZManaged[R, E, (ZStream[R,E,A],ZStream[R,E,A])] = {
```

Comment:
Instead of the overload, let's use a default parameter:
`def split[B1, B2](p: A => UIO[Boolean], queueCapacity: Int = 16): ZManaged[R, E, (ZStream[R,E,A],ZStream[R,E,A])]`

Reply:
Also, please add scaladoc :-)

Reply:
We should also call it `partition` IMO.